### PR TITLE
Remove vending useAsync calls and remove it all together

### PR DESCRIPTION
**Pull request by razzeee**
@kymckay Can you test this, I've done this mostly blind now, due to not having a vending setup here.

I think we'll also need to talk about `vending.ts`, as it seems it had some logic e.g. `DEFAULT_SETUP`, that should be part of the backend and not in the client.

### CODE REVIEW

File: frontend/src/components/application/AppVendingControls/OwnershipTokens/TokenCancelButton.tsx [hunk 1]
```diff
@@ -20,43 +20,41 @@ const TokenCancelButton: FunctionComponent<Props> = ({
...
+    },
+    onSuccess: (data) => {
+      if (data.data[0].status === "cancelled") {
+        setState("cancelled")
+        toast.success(t("token-cancelled"))
+      }
+    },
+    onError: (error) => {
+      toast.error(t(error as string))
+    },
```
Comment (kymckay) [inline frontend/src/components/application/AppVendingControls/OwnershipTokens/TokenCancelButton.tsx:41]: Will give this PR a test tomorrow morning and just merge if it's good :+1:

General note: I believe `onSuccess` and `onError` are deprecated. Same effects can be achieved inside the query function.

Comment (razzeee) [reply frontend/src/components/application/AppVendingControls/OwnershipTokens/TokenCancelButton.tsx:41]: Yes, they are deprecated, but I wasn't sure if that applies to mutations too. But it probably does.

Comment (razzeee) [reply frontend/src/components/application/AppVendingControls/OwnershipTokens/TokenCancelButton.tsx:41]: Also, we might want to address the `DEFAULT_SETUP` problem first.

Comment (razzeee) [reply frontend/src/components/application/AppVendingControls/OwnershipTokens/TokenCancelButton.tsx:41]: Then again, this isn't used, so creating an issue for that would also be fine in my book.

**Comment by razzeee** [comment]
Working on this, also made it obvious, that the created names for the endpoints are horrible. I can only figure out what is what by knowing if it's post or get and looking at the params, because the names are soooo cryptic and almost have everything in them.

**Comment by kymckay** [comment]
Got distracted, but testing this now

**Comment by razzeee** [comment]
Followup in https://github.com/flathub/website/pull/2202

**Comment by kymckay** [comment]
Annoyingly Stripe have added additional steps to Express account creation which serve as blockers in test mode 
image

Making payments seem to be working though. I'm getting `invalid-appid` errors when making tokens and accessing payment settings - which may be related to the lack of express account (we should be handling that better regardless).

**Comment by razzeee** [comment]
Can you test if you run into these without these changes?

**Pull request closed without merging**